app/cloud_storage.py

```diff
@@ -0,0 +1,155 @@
+            media_body=media).execute()
+
+    def compose_objects(
+        self,
+        source_object_names,
+        destination_object_name,
+        destination_content_type):
+        """Concatenates the source objects to generate the destination object.
+        
+        Args:
+            source_object_names (list of str): Names of the source objects.
+            destination_object_name (str): Name of the destination object.
+            destination_content_type (str): MIME type of the destination object.
+        """
+        self.service.objects().compose(
+            destinationBucket=self.bucket_name,
+            destinationObject=destination_object_name,
+            body={
+                'sourceObjects': [{'name': name} for name in source_object_names],
+                'destination': {
+                    'contentType': destination_content_type,
+                    # This let browsers to download the file instead of opening
+                    # it in a browser.
+                    'contentDisposition':
+                       'attachment; filename=%s' % destination_object_name,
+                },
+            }).execute()
+
+    def sign_url(self, object_name):
+        """ Generates Cloud Storage signed URL to download Google Cloud Storage
+        object without sign in.
+
+        See: https://cloud.google.com/storage/docs/access-control/signed-urls
+        
+        This only works on a real App Engine app, not in a dev app server.
+        
+        Args:
+            object_name (str): The name of the object which is signed.
+        """
+        if utils.is_dev_app_server():
+            # Not working on a dev app server because it doesn't support
+            # app_identity.sign_blob(). An alternative implementation would
+            # be needed to make it work on a dev app server.
+            raise Exception(
+                'sign_url only works on a real App Engine app, not on a dev '
+                'app server.')
+
+        method = 'GET'
+        expiration_time = utils.get_utcnow() + SIGNED_URL_LIFETIME
+        expiration_sec = int(time.mktime(expiration_time.timetuple()))
+        path = '/%s/%s' % (self.bucket_name, object_name)
+
+        # These are unused in our use case.
+        content_md5 = ''
+        content_type = ''
+
+        signed_text = '\n'.join([
+            method,
+            content_md5,
+            content_type,
+            str(expiration_sec),
+            path,
+        ])
+        (_, signature) = app_identity.sign_blob(signed_text)
+
+        query_params = {
+            'GoogleAccessId': app_identity.get_service_account_name(),
+            'Expires': str(expiration_sec),
+            'Signature': base64.b64encode(signature),
+        }
+        return 'https://storage.googleapis.com%s?%s' % (path, urllib.urlencode(query_params))
+
+    def set_objects_lifetime(self, lifetime_days):
+        """Sets lifetime of all objects in the bucket in days.
```

Comment:
> AFAIK we're not using Cloud Storage for anything else (right?)

Right.

> but it might be good to add a comment (maybe around line 38, where it says the default bucket is used) warning that objects in that bucket have a fixed lifetime and we'd need to split things into separate buckets if that becomes a problem.

Done. looks there's also a way to extend lifetime for specific objects by setting [object holds](https://cloud.google.com/storage/docs/bucket-lock#object-holds).